src/wizard/wizard-dialog.ts

```diff
@@ -227,24 +227,32 @@ export class ESPHomeWizardDialog extends LitElement {
     let hideActions = false;
     const content = html`
       <div>
-        A device needs to be connected to a computer using a USB cable to be
-        added to ESPHome. Once added, ESPHome will interact with the device
+        A device needs to be connected to a computer (either the server running
+        ESPHome or the client running the browser) using a USB cable to be added
+        to ESPHome. Once added, ESPHome will interact with the device
         wirelessly.
       </div>
       <div>
-        ${allowsWebSerial
-          ? "Your browser does not support WebSerial."
-          : "You are not browsing the ESPHome Device Builder over a secure connection (HTTPS)."}
-        This prevents ESPHome from being able to install this on devices
-        connected to this computer.
+        The following issues prevent ESPHome from being able to remotely install the necessary firmware on
+        devices physically connected to the client computer (running the browser):
+        <ul>
+          <li>Your browser does not support WebSerial.</li>
+          ${allowsWebSerial
```

Comment:
Note that `allowsWebSerial` is somewhat confusingly named which might make this logic a bit confusing to read. `allowsWebSerial` tells us whether user is running in a secure context (whether `window.isSecureContext` is set).

This dialog is only shown when WebSerial is not supported so updated message always shows the `Your browser does not support WebSerial.` message unlike previous logic that has shown either this message or the one about not running in secure context.